src/components/CardItem.js

```diff
@@ -0,0 +1,82 @@
+import { useEffect, useState } from "react";
+import { getFolder } from "./Api";
+import noImage from "../assets/noimage.jpeg";
+
+function getTimeAgo(createdAt) {
+  const now = new Date();
+  const createdDate = new Date(createdAt);
+
+  const timeDiff = now - createdDate;
+  const minutes = Math.floor(timeDiff / (1000 * 60));
+  const hours = Math.floor(timeDiff / (1000 * 60 * 60));
+  const days = Math.floor(timeDiff / (1000 * 60 * 60 * 24));
+  const months = Math.floor(timeDiff / (1000 * 60 * 60 * 24 * 30));
+  const years = Math.floor(timeDiff / (1000 * 60 * 60 * 24 * 365));
+
+  if (minutes < 2) {
+    return "1 minute ago";
+  } else if (minutes <= 59) {
+    return `${minutes} minutes ago`;
+  } else if (hours < 1) {
+    return "1 hour ago";
+  } else if (hours <= 23) {
+    return `${hours} hours ago`;
+  } else if (days < 1) {
+    return "1 day ago";
+  } else if (days <= 30) {
+    return `${days} days ago`;
+  } else if (months < 1) {
+    return "1 month ago";
+  } else if (months <= 11) {
+    return `${months} months ago`;
+  } else if (years < 1) {
+    return "1 year ago";
+  } else {
+    const roundedYears = Math.floor(months / 12);
+    return `${roundedYears} years ago`;
+  }
+}
+
+function CardItem() {
+  const [cardData, setCardData] = useState();
+
+  useEffect(() => {
+    const getFolderData = async () => {
+      try {
+        const body = await getFolder();
+        setCardData(body.folder.links);
+      } catch (error) {
+        console.log(error);
+      }
+    };
+
+    getFolderData();
+  }, []);
+
+  return cardData ? (
+    cardData.map((data) => (
+      <a href={data.url} className="CardItem" key={data.id} target="_blank">
+        <img
+          className="cardImg"
+          src={data.imageSource || noImage}
+          alt="링크 대표 이미지"
+        />
+        <div className="descriptionBox">
+          <p className="cardTime">{getTimeAgo(data.createdAt)}</p>
+          <p className="description">
+            {data.title}
+            <br />
+            {data.description}
+          </p>
+          <p className="cardDate">
+            {new Date(data.createdAt).toLocaleDateString()}
+          </p>
+        </div>
+      </a>
+    ))
+  ) : (
+    <div>Loading...</div>
+  );
```

Comment:
- cardData state의 초기값이 지정되어있지 않기 때문에 초기 렌더링 시 undefined가 됩니다.
때문에 조건이 없었다면 `cardData.map` => `undefined.map`이어서 `cannot read properties on undefined(reading map)`이라는 에러가 발생했을 거예요!
위 리뷰처럼 cardData state에 초기값으로 빈 배열을 주시면 오류는 발생하지 않습니다.
- 대신 `Loading...` 화면을 보여주려면 조건이 바뀌어야겠죠?
이 경우, 일반적으로 early return하여 다음과 같이 구현할 수 있습니다.
그러나 cardList가 비어있다고 해서 Loading 상태인 건 아니기 때문에, 대신 '카드 리스트가 비어 있음'을 표출해주는 게 더 명확하겠네요.
```suggestion
  const isListEmpty = cardList.length === 0;
  
  if (isListEmpty) {
    return <div>목록이 비어 있습니다.</div>
  }
  
  return ...
```